src/Microsoft.AspNet.TestHost/WebSocket.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.TestHost
+{
+    class WebSocket : System.Net.WebSockets.WebSocket
```

Comment:
It's not the same code, it's only 'inspired by' the websocket middleware. It cuts out the wire-serialization layer and only passes messages back and forth. This should make the test scenarios simpler and easier to debug.